Foundation/NSString.swift

```diff
@@ -1234,8 +1231,8 @@ extension NSString {
     }
     
     public convenience init?(bytes: UnsafeRawPointer, length len: Int, encoding: UInt) {
-        let bytePtr = bytes.bindMemory(to: UInt8.self, capacity: len)
-        guard let cf = CFStringCreateWithBytes(kCFAllocatorDefault, bytePtr, len, CFStringConvertNSStringEncodingToEncoding(encoding), true) else {
+        let extRep = NSData(bytesNoCopy: UnsafeMutableRawPointer(mutating: bytes), length: len, freeWhenDone: false)._cfObject
```

Comment:
That's interesting. I'm not in front of a computer at the moment, but IIRC, testing on OS X showed that Apple Foundation `NSString.init(bytes:length:encoding:)` proceeded past an internal null character. I will check again later in the day and report back.